velox/expression/CastExpr-inl.h

```diff
@@ -301,6 +304,78 @@ VectorPtr CastExpr::applyDecimalToBooleanCast(
   return result;
 }
 
+template <typename FromNativeType>
+VectorPtr CastExpr::applyDecimalToVarcharCast(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& fromType) {
+  VectorPtr result;
+  context.ensureWritable(rows, VARCHAR(), result);
+  (*result).clearNulls(rows);
+  const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
+  int precision = getDecimalPrecisionScale(*fromType).first;
+  int scale = getDecimalPrecisionScale(*fromType).second;
+  // A varchar's size is estimated with unscaled value digits, dot, leading
+  // zero, and possible minus sign.
+  int varcharSize = precision + !!(scale > 0) + !!(precision == scale) + 1;
+  // Calculate the max total size of the raw string buffer.
+  size_t maxTotalResultBytes = rows.countSelected() * varcharSize;
+
+  auto flatResult = result->asFlatVector<StringView>();
+  Buffer* buffer = flatResult->getBufferWithSpace(maxTotalResultBytes);
+  char* rawBuffer = buffer->asMutable<char>() + buffer->size();
+
+  applyToSelectedNoThrowLocal(context, rows, result, [&](vector_size_t row) {
+    if (simpleInput->isNullAt(row)) {
+      result->setNull(row, true);
+    } else {
+      auto unscaledValue = simpleInput->valueAt(row);
+      const char* startPosition = rawBuffer;
+      if (unscaledValue == 0) {
+        *rawBuffer++ = '0';
+      } else {
+        if (unscaledValue < 0) {
+          *rawBuffer++ = '-';
+          unscaledValue = -unscaledValue;
+        }
+        auto [position, ec] = std::to_chars(
+            rawBuffer,
+            rawBuffer + varcharSize,
+            unscaledValue / DecimalUtil::kPowersOfTen[scale]);
+        VELOX_DCHECK(
+            ec == std::errc(),
+            "Cast from decimal to varchar fails because {}.",
+            std::make_error_code(ec).message());
+        rawBuffer = position;
+
+        if (scale > 0) {
+          *rawBuffer++ = '.';
+          uint128_t fraction = unscaledValue % DecimalUtil::kPowersOfTen[scale];
+          // Append leading zeros.
+          int numOfZeros =
+              std::max(scale - fmt::detail::count_digits(fraction), 0);
+          std::memset(rawBuffer, '0', numOfZeros);
+          rawBuffer += numOfZeros;
+          // Append remaining fraction digits.
+          auto [position, ec] =
+              std::to_chars(rawBuffer, rawBuffer + varcharSize, fraction);
+          VELOX_DCHECK(
+              ec == std::errc(),
+              "Cast from decimal to varchar fails because {}.",
+              std::make_error_code(ec).message());
+          rawBuffer = position;
+        }
+      }
+      flatResult->setNoCopy(
```

Comment:
If the StringView is inline
```
flatResult->setNoCopy(row, StringView(startPosition, rawBuffer - startPosition)))
rawBuffer = startPosition;

```
We don't need to use the buffer allocated from `stringValues`.

Reply:
Updated.